compiler/rustc_builtin_macros/src/format_foreign.rs

```diff
@@ -421,7 +423,10 @@ pub(crate) mod printf {
                             state = Prec;
                             parameter = None;
                             flags = "";
-                            width = Some(Num::from_str(at.slice_between(end).unwrap(), None));
+                            width = Num::from_str(at.slice_between(end).unwrap(), None);
```

Comment:
```suggestion
                            width = at.slice_between(end).map(|num| Num::from_str(num, None));
```

Reply:
should add `?` at end to convert `Option<Option<Num>>` to `Option<Num>`